src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -36,31 +36,24 @@ internal static bool IsEnabled()
             //HttpClientHandler is responsible to call DiagnosticsHandler.IsEnabled() before forwarding request here.
             //This code will not be reached if no one listens to 'HttpHandlerDiagnosticListener', unless consumer unsubscribes
             //from DiagnosticListener right after the check. So some requests happening right after subscription starts
-            //might not be instrumented. Similarly, when consumer unsubscribes, extra requests might be instumented
+            //might not be instrumented. Similarly, when consumer unsubscribes, extra requests might be instrumented
 
             if (request == null)
             {
                 throw new ArgumentNullException(nameof(request), SR.net_http_handler_norequest);
             }
 
-            Activity activity = null;
-            Guid loggingRequestId = Guid.Empty;
+            Activity activity = s_diagnosticListener.StartActivityIfEnabled(
+                () => new Activity(DiagnosticsHandlerLoggingStrings.ActivityName), 
+                request,
+                null,
+                () => new { Request = request}
```

Comment:
Note that creating this function actually causes two allocations.   One for the closure that captures 'request' and one for the delegate that points at that closure.    

The sad fact is that closures almost never help in micro-optimization (avoiding allocations), unless they capture NO variables (you pass them through the arguments of the Func.